storage/file_persister.go

```diff
@@ -43,3 +48,142 @@ func (l *LocalFilePersister) Persist(path string, data io.Reader) (err error) {
 
 	return nil
 }
+
+// RemoteFilePersister is to be used when files created by the browser module need
+// to be uploaded to a remote location. This uses a preSignedURLGetterURL to
+// retrieve one pre-signed URL. The pre-signed url is used to upload the file
+// to the remote location.
+type RemoteFilePersister struct {
+	preSignedURLGetterURL string
+	headers               map[string]string
+	basePath              string
+}
+
+// Persist will upload the contents of data to a remote location.
+func (r *RemoteFilePersister) Persist(ctx context.Context, path string, data io.Reader) (err error) {
+	pURL, err := r.getPreSignedURL(ctx, path)
+	if err != nil {
+		return fmt.Errorf("get presigned url: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPut, pURL, data)
+	if err != nil {
+		return fmt.Errorf("create upload request: %w", err)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("perform upload request: %w", err)
+	}
+	defer func() {
+		if _, cerr := io.Copy(io.Discard, resp.Body); cerr != nil && err == nil {
+			err = fmt.Errorf("drain upload response body: %w", cerr)
+		}
+
+		if cerr := resp.Body.Close(); cerr != nil && err == nil {
+			err = fmt.Errorf("close upload response body: %w", cerr)
+		}
```

Comment:
I don't think we should check the error from `Close`. `nolint` here is fine.

Reply:
If this was in a test, i would agree. What's the reason for not checking?

Reply:
Yep, that makes sense since we can't do much if it fails 👍 Resolved in 4e35b0857d614e2ced20499457bc30cc3ebd21f3